test/integration-no-mock/config-management/env-config-export.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2018, Kinvey, Inc. All rights reserved.
+ *
+ * This software is licensed to you under the Kinvey terms of service located at
+ * http://www.kinvey.com/terms-of-use. By downloading, accessing and/or using this
+ * software, you hereby accept such terms of service  (and any agreement referenced
+ * therein) and agree that you have read, understand and agree to be bound by such
+ * terms of service and are of legal age to agree to such terms with Kinvey.
+ *
+ * This software contains valuable confidential and proprietary information of
+ * KINVEY, INC and is subject to applicable licensing agreements.
+ * Unauthorized reproduction, transmission or distribution of this file and its
+ * contents is a violation of applicable laws.
+ */
+
+const async = require('async');
+const moment = require('moment');
+const path = require('path');
+
+const ApiService = require('./../../ApiService');
+const ConfigManagementHelper = require('./../../ConfigManagementHelper');
+const { execCmdWoMocks, randomStrings } = require('./../../TestsHelper');
+
+module.exports = () => {
+  const appName = randomStrings.appName();
+
+  before('setup app', (done) => {
+    ConfigManagementHelper.app.createInTestsOrg({ name: appName }, done);
+  });
+
+  after('remove app', (done) => {
+    execCmdWoMocks(`app delete ${appName} --no-prompt`, null, done);
+  });
+
+  it('with collections, hooks, endpoints, push settings, groups, roles should succeed', (done) => {
```

Comment:
The test does not fail if the custom endpoints, push settings, groups or roles are not exported correctly. It seems that the problem is in the used assertion function - _ConfigManagementHelper.env.assertExportedConfig_

Reply:
The roles are still not validated

Reply:
Yes, this is intentional. I've found a bug and logged it (cli-135). Until it is fixed, they will fail and I decided to skip them and add a TODO there.